icecream/icecream.py

```diff
@@ -154,15 +156,40 @@ def format_pair(prefix, arg, value):
     return '\n'.join(lines)
 
 
-def argumentToString(obj):
-    s = DEFAULT_ARG_TO_STRING_FUNCTION(obj)
+# def argumentToString(obj, width):
+def argumentToString(obj, width=DEFAULT_LINE_WRAP_WIDTH):
+    s = DEFAULT_ARG_TO_STRING_FUNCTION(obj, width=width)
     s = s.replace('\\n', '\n')  # Preserve string newlines in output.
     return s
 
 
+def columns():
+    """ Returns the number of columns that this terminal can handle. """
+    width = DEFAULT_LINE_WRAP_WIDTH
+    try:
+        # TODO come up with a more elegant solution than subtracting
+        #  a seemingly random number of characters.
```

Comment:
Any thoughts about this overflow thingy? I just realised that the extra characters might come from the `ic |` prefix.

Reply:
That's probably what it is, yes. Have this function return the full width, then deal with the length of the prefix inside the class. Remember there's both the `ic |` prefix (which can be configured) as well as the `argPrefix`.

Reply:
Let me know what you think of my new solution.